packages/block-editor/src/components/observe-typing/index.js

```diff
@@ -53,58 +60,60 @@ function isKeyDownEligibleForStartTyping( event ) {
  * element.
  */
 export function useMouseMoveTypingReset() {
-	const isTyping = useSelect(
-		( select ) => select( blockEditorStore ).isTyping(),
-		[]
-	);
-	const { stopTyping } = useDispatch( blockEditorStore );
+	const { isTyping } = useSelect( blockEditorStore );
+	const { stopTyping, startTyping } = useDispatch( blockEditorStore );
 
-	return useRefEffect(
-		( node ) => {
-			if ( ! isTyping ) {
-				return;
-			}
+	return useRefEffect( ( node ) => {
+		const { ownerDocument } = node;
+		let lastClientX;
+		let lastClientY;
+		let timeoutSinceLastMouseMove;
 
-			const { ownerDocument } = node;
-			let lastClientX;
-			let lastClientY;
+		/**
+		 * On mouse move, unset typing flag if user has moved cursor.
+		 *
+		 * @param {MouseEvent} event Mousemove event.
+		 */
+		function stopTypingOnMouseMove( event ) {
+			const { clientX, clientY } = event;
 
-			/**
-			 * On mouse move, unset typing flag if user has moved cursor.
-			 *
-			 * @param {MouseEvent} event Mousemove event.
-			 */
-			function stopTypingOnMouseMove( event ) {
-				const { clientX, clientY } = event;
+			// We need to check that the mouse really moved because Safari
+			// triggers mousemove events when shift or ctrl are pressed.
+			const didMove =
+				lastClientX &&
+				lastClientY &&
+				( lastClientX !== clientX || lastClientY !== clientY );
 
-				// We need to check that the mouse really moved because Safari
-				// triggers mousemove events when shift or ctrl are pressed.
-				if (
-					lastClientX &&
-					lastClientY &&
-					( lastClientX !== clientX || lastClientY !== clientY )
-				) {
-					stopTyping();
-				}
+			if ( isTyping() && didMove ) {
```

Comment:
```suggestion
			if ( didMove && isTyping() ) {
```

Avoids calling a selector if there's no need for it. Obsessing over the little things. :)